File: sorting/quick_sort/quick_sort.c
#include <stdio.h>

void quickSort(int in[], int low, int high) {
  int i, j, pivot, tmp;

  if (low < high) {
    pivot = low;
    i = low;
    j = high;

    while (i < j) {
      while (in[i] <= in[pivot] && i < high) {
        i++;
      }
      while (in[j] > in[pivot]) {
        j--;
      }
      if (i < j) {
        tmp = in[i];
        in[i] = in[j];
        in[j] = tmp;
      }
    }
    tmp = in[pivot];
    in[pivot] = in[j];
    in[j] = tmp;
    quickSort(in, low, j - 1);
    quickSort(in, j + 1, high);
  }
}

int main() {
  int in[] = {100, 1, 44, 22, 55, 323, 92, 12, 15, 48};
  int count = sizeof(in) / sizeof(in[0]);
  quickSort(in, 0, count);

  for (int i = 0; i < count; i++) {
    printf("Value %d\n", in[i]);
  }
}

File: sorting/quick_sort/quick_sort.go
package main

import (
    "fmt"
)

func partition(in []int, low, high int) int {
    pivot := in[high]
    for j := low; j < high; j++ {
     if in[j] < pivot {
         //Swap
        in[j],in[low] = in[low],in[j]
        low++
     }
    }
    in[low],in[high] = in[high],in[low]
    return low
}

func quickSort(in []int, low, high int){
    if low > high {
        return
    }

    part := partition(in, low,high)
    quickSort(in, low, part - 1)
    quickSort(in, part+1, high)
}

func main(){
    in := []int{12,44,13,1,99,22,0,200,8391,461}
    quickSort(in, 0, len(in) - 1)
    for _,v := range in {
        fmt.Printf("Value %d\n",v)
    }
}
